x/staking/keeper/cons_pubkey.go

```diff
@@ -90,3 +132,88 @@ func bytesSliceExists(sliceList [][]byte, targetBytes []byte) bool {
 	}
 	return false
 }
+
+// UpdateAllMaturedConsKeyRotatedKeys udpates all the matured key rotations.
+func (k Keeper) UpdateAllMaturedConsKeyRotatedKeys(ctx sdk.Context, maturedTime time.Time) error {
+	maturedRotatedValAddrs, err := k.GetAllMaturedRotatedKeys(ctx, maturedTime)
+	if err != nil {
+		return err
+	}
+
+	for _, valAddr := range maturedRotatedValAddrs {
+		err := k.deleteConsKeyIndexKey(ctx, valAddr, maturedTime)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// deleteConsKeyIndexKey deletes the key which is formed with the given valAddr, time.
+func (k Keeper) deleteConsKeyIndexKey(ctx sdk.Context, valAddr sdk.ValAddress, ts time.Time) error {
+	rng := new(collections.Range[collections.Pair[[]byte, time.Time]]).
+		EndInclusive(collections.Join(valAddr.Bytes(), ts))
```

Comment:
this will remove all from start up to `valAddr, ts`.

this means that if we have in state:

`val1-12:00`
`val1-13:00`
`val2-15:00`

if i provide val2-15:00
then also `val1:12:00` and `val1:13:00`.

If this is not expected and no test is failing then we should:
1. add a test whcih asserts the bug
2. fix the bug here by using a different range.

Reply:
done!
https://github.com/cosmos/cosmos-sdk/pull/18236/commits/75eb409b0201993d33f7daef691c656cea90a4e0